### PR TITLE
First functional version of symbolic abstraction for timed hybrid systems

### PR DESCRIPTION
**Pull request by CanFlo**
This pull request introduces a first functional version of symbolic abstraction for timed hybrid systems, including:
• A new module: TimedHybridAutomata
• Base modules: VectorContinuousSystem and TimeSymbolicModel (required for TimedHybridAutomata construction)
• Three test suites (one per module)
• The ability to generate and manipulate symbolic abstractions for non-trivial hybrid systems (see test_TimedHybridAutomata.jl)

⚠️ This is not the final version: the API, robustness, and flexibility will still be improved, but the current structure already supports interesting case studies.

1) VectorContinuousSystem
• Introduces a generic container to aggregate multiple continuous subsystems (state, time, etc.) within a single mode
• Allows easy modeling of hybrid systems where each mode combines several components (e.g., physical dynamics + clock)

2) TimeSymbolicModel
• Adds a symbolic model for the temporal component of each mode
• Provides utilities to retrieve time indices, extract intervals, etc.

3) TimedHybridAutomata
• New central module for building symbolic timed hybrid automata

4) test_TimedHybridAutomata.jl
• Test 1: Simple hybrid system with time
◦ Two modes, 1D dynamics, transition with fixed reset map
◦ Checks structure, mappings, transitions, utilities, error cases

• Test 2: Simple hybrid system with frozen time
◦ Two modes, time not tracked (time-frozen)
◦ Checks that all states have time_index=1, and transitions do not affect time

• Test 3: Hybrid system with 3 modes and full transitions
◦ Three modes, 2D dynamics, 6 transitions with permutations, offsets, tmin, bounding
◦ Robustness checks: structure, mappings, transitions, reset maps, utilities, edge cases, global consistency

Points for improvement / Next steps
• Allow users to provide custom abstraction methods and optimizers for each mode
• Improve reset map handling at cell boundaries (some edge cases not yet handled)
• Extend test coverage to other use cases (higher dimensions, uncertainties, etc.)
• Optimize performance and modularity (e.g., shared symbolic models across modes)
• Later: enable solving problems directly on these abstractions (part 2)

### CODE REVIEW

File: src/symbolic/HybridTimedAutomata/TimeSymbolicModel.jl [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+Structure representing the symbolic abstraction of time for a mode in a hybrid system.
+
+# Fields
+- `tsteps::Vector{Float64}`: Discrete time steps
+- `time_domain::Any`: The time domain (e.g., a HyperRectangle)
+- `is_active::Bool`: True if time evolves, false if time is frozen
+"""
+struct TimeSymbolicModel
+    tsteps::Vector{Float64}
+    time_domain::Any
```
Comment (blegat) [inline src/symbolic/HybridTimedAutomata/TimeSymbolicModel.jl:14]: Why don't we add the type of `time_domain` to the `struct` like `struct TimeSymbolicModel{T}; time_domain::T` ? Is the access to this field critical for performance ? See https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type

Comment (CanFlo) [reply src/symbolic/HybridTimedAutomata/TimeSymbolicModel.jl:14]: Yes, I will also remove mutable from the struct definition of TimedHybridAutomata to follow Julia best practices, since the fields are not meant to be modified after initialization.

File: src/symbolic/HybridTimedAutomata/VectorContinuousSystem.jl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+@warn "This type of mathematical system is necessary to perform the Cartesian product between the dynamical system and time. Eventually, this code should be integrated into MathematicalSystems.jl and should no longer be present in the Dionysos codebase, or at least not in the symbolic part."
```
Comment (blegat) [inline src/symbolic/HybridTimedAutomata/VectorContinuousSystem.jl:2]: Just make it into a comment, no need for this to display a warning when loading Dionysos

File: src/symbolic/HybridTimedAutomata/TimedHybridAutomata.jl [hunk 3]
```diff
@@ -0,0 +1,645 @@
+export TimedHybridAutomata
+
+module TimedHybridAutomata
+
+using HybridSystems, MathematicalSystems, StaticArrays, JuMP
```
Comment (blegat) [inline src/symbolic/HybridTimedAutomata/TimedHybridAutomata.jl:5]: Prefer using `import` inside the `src` module. `using` is fine in the test and docs but in the code, stick to import

Comment (blegat) [reply src/symbolic/HybridTimedAutomata/TimedHybridAutomata.jl:5]: Cfr https://jump.dev/JuMP.jl/stable/developers/style/#using-vs.-import

Comment (blegat) [reply src/symbolic/HybridTimedAutomata/TimedHybridAutomata.jl:5]: By the way, you can probably replace `using JuMP` by `import MathOptInterface as MOI`

**Comment by CanFlo** [comment]
> Is this still a draft or is it ready for review ?

Not currently ready

File: src/optim/abstraction/timed_hybrid_abstraction.jl [hunk 4]
```diff
@@ -0,0 +1,711 @@
...
+- `concret_cost_fun::F`: Concrete cost function for optimal control
+- `problem_type::Symbol`: `:optimal_control` or `:safety`
+- `time_horizon::Union{Real, Dionysos.Problem.Infinity}`: Time horizon constraint
+"""
+struct TimedHybridProblemSpecs{F} # Need to add initial region for safety problems (?)
+    initial_state::Tuple{AbstractVector{Float64}, Float64, Int} # aug_state initial ([x], t, mode_id)
+    Xs_target::Vector{<:Dionysos.Utils.HyperRectangle} # target sets or safe sets depending on problem type
+    Ts_target::Vector{<:Dionysos.Utils.HyperRectangle}
+    Ns_target::Vector{Int}
+    concret_cost_fun::F
```
Comment (blegat) [inline src/optim/abstraction/timed_hybrid_abstraction.jl:41]:
```suggestion
    concrete_cost_fun::F
```

Comment (CanFlo) [reply src/optim/abstraction/timed_hybrid_abstraction.jl]: fixed

**Comment by CanFlo** [comment]
Where should I move 
- Dionysos.Optim.Abstraction.TimedHybridAbstraction.TimedHybridProblemSpecs
- Dionysos.Optim.Abstraction.TimedHybridAbstraction.TimedHybridOptimalControlProblem
- Dionysos.Optim.Abstraction.TimedHybridAbstraction.TimedHybridSafetyProblem
- Dionysos.Optim.Abstraction.TimedHybridAbstraction.solve_timed_hybrid_problem
- Dionysos.Optim.Abstraction.TimedHybridAbstraction.get_closed_loop_trajectory 

It seems that it should not be in Optim
